model/model.go

```diff
@@ -18,14 +18,20 @@ import (
 	"fmt"
 	"strconv"
 	"strings"
+	"sync"
+
+	"github.com/casbin/casbin/v3/rbac"
 
 	"github.com/casbin/casbin/v3/config"
 	"github.com/casbin/casbin/v3/log"
 	"github.com/casbin/casbin/v3/util"
 )
 
 // Model represents the whole access control model.
-type Model map[string]AssertionMap
+type Model struct {
+	Data  map[string]AssertionMap
```

Comment:
`Data` and `Mutex` should be private.

Reply:
We should avoid users using `Data` directly.

Reply:
I made them private, but in order to get policy when enforce, I added `GetAssertion()`. But this may also allow users to operate `Assertion`, what are your suggestions?